File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/UserItemDataDto.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.IOException;
import org.threeten.bp.OffsetDateTime;
/**
 * Class UserItemDataDto  
 */
@Schema(description = "Class UserItemDataDto  ")

public class UserItemDataDto {
  @SerializedName("Rating")
  private Double rating = null;

  @SerializedName("PlayedPercentage")
  private Double playedPercentage = null;

  @SerializedName("UnplayedItemCount")
  private Integer unplayedItemCount = null;

  @SerializedName("PlaybackPositionTicks")
  private Long playbackPositionTicks = null;

  @SerializedName("PlayCount")
  private Integer playCount = null;

  @SerializedName("IsFavorite")
  private Boolean isFavorite = null;

  @SerializedName("LastPlayedDate")
  private OffsetDateTime lastPlayedDate = null;

  @SerializedName("Played")
  private Boolean played = null;

  @SerializedName("Key")
  private String key = null;

  @SerializedName("ItemId")
  private String itemId = null;

  @SerializedName("ServerId")
  private String serverId = null;

  public UserItemDataDto rating(Double rating) {
    this.rating = rating;
    return this;
  }

   /**
   * The rating.
   * @return rating
  **/
  @Schema(description = "The rating.")
  public Double getRating() {
    return rating;
  }

  public void setRating(Double rating) {
    this.rating = rating;
  }

  public UserItemDataDto playedPercentage(Double playedPercentage) {
    this.playedPercentage = playedPercentage;
    return this;
  }

   /**
   * The played percentage.
   * @return playedPercentage
  **/
  @Schema(description = "The played percentage.")
  public Double getPlayedPercentage() {
    return playedPercentage;
  }

  public void setPlayedPercentage(Double playedPercentage) {
    this.playedPercentage = playedPercentage;
  }

  public UserItemDataDto unplayedItemCount(Integer unplayedItemCount) {
    this.unplayedItemCount = unplayedItemCount;
    return this;
  }

   /**
   * The unplayed item count.
   * @return unplayedItemCount
  **/
  @Schema(description = "The unplayed item count.")
  public Integer getUnplayedItemCount() {
    return unplayedItemCount;
  }

  public void setUnplayedItemCount(Integer unplayedItemCount) {
    this.unplayedItemCount = unplayedItemCount;
  }

  public UserItemDataDto playbackPositionTicks(Long playbackPositionTicks) {
    this.playbackPositionTicks = playbackPositionTicks;
    return this;
  }

   /**
   * The playback position ticks.
   * @return playbackPositionTicks
  **/
  @Schema(description = "The playback position ticks.")
  public Long getPlaybackPositionTicks() {
    return playbackPositionTicks;
  }

  public void setPlaybackPositionTicks(Long playbackPositionTicks) {
    this.playbackPositionTicks = playbackPositionTicks;
  }

  public UserItemDataDto playCount(Integer playCount) {
    this.playCount = playCount;
    return this;
  }

   /**
   * The play count.
   * @return playCount
  **/
  @Schema(description = "The play count.")
  public Integer getPlayCount() {
    return playCount;
  }

  public void setPlayCount(Integer playCount) {
    this.playCount = playCount;
  }

  public UserItemDataDto isFavorite(Boolean isFavorite) {
    this.isFavorite = isFavorite;
    return this;
  }

   /**
   * A value indicating whether this instance is favorite.
   * @return isFavorite
  **/
  @Schema(description = "A value indicating whether this instance is favorite.")
  public Boolean isIsFavorite() {
    return isFavorite;
  }

  public void setIsFavorite(Boolean isFavorite) {
    this.isFavorite = isFavorite;
  }

  public UserItemDataDto lastPlayedDate(OffsetDateTime lastPlayedDate) {
    this.lastPlayedDate = lastPlayedDate;
    return this;
  }

   /**
   * The last played date.
   * @return lastPlayedDate
  **/
  @Schema(description = "The last played date.")
  public OffsetDateTime getLastPlayedDate() {
    return lastPlayedDate;
  }

  public void setLastPlayedDate(OffsetDateTime lastPlayedDate) {
    this.lastPlayedDate = lastPlayedDate;
  }

  public UserItemDataDto played(Boolean played) {
    this.played = played;
    return this;
  }

   /**
   * A value indicating whether this &#x60;MediaBrowser.Model.Dto.UserItemDataDto&#x60; is played.
   * @return played
  **/
  @Schema(description = "A value indicating whether this `MediaBrowser.Model.Dto.UserItemDataDto` is played.")
  public Boolean isPlayed() {
    return played;
  }

  public void setPlayed(Boolean played) {
    this.played = played;
  }

  public UserItemDataDto key(String key) {
    this.key = key;
    return this;
  }

   /**
   * The key.
   * @return key
  **/
  @Schema(description = "The key.")
  public String getKey() {
    return key;
  }

  public void setKey(String key) {
    this.key = key;
  }

  public UserItemDataDto itemId(String itemId) {
    this.itemId = itemId;
    return this;
  }

   /**
   * The item identifier.
   * @return itemId
  **/
  @Schema(description = "The item identifier.")
  public String getItemId() {
    return itemId;
  }

  public void setItemId(String itemId) {
    this.itemId = itemId;
  }

  public UserItemDataDto serverId(String serverId) {
    this.serverId = serverId;
    return this;
  }

   /**
   * Used only by our Windows app. Not used by Emby Server.
   * @return serverId
  **/
  @Schema(description = "Used only by our Windows app. Not used by Emby Server.")
  public String getServerId() {
    return serverId;
  }

  public void setServerId(String serverId) {
    this.serverId = serverId;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    UserItemDataDto userItemDataDto = (UserItemDataDto) o;
    return Objects.equals(this.rating, userItemDataDto.rating) &&
        Objects.equals(this.playedPercentage, userItemDataDto.playedPercentage) &&
        Objects.equals(this.unplayedItemCount, userItemDataDto.unplayedItemCount) &&
        Objects.equals(this.playbackPositionTicks, userItemDataDto.playbackPositionTicks) &&
        Objects.equals(this.playCount, userItemDataDto.playCount) &&
        Objects.equals(this.isFavorite, userItemDataDto.isFavorite) &&
        Objects.equals(this.lastPlayedDate, userItemDataDto.lastPlayedDate) &&
        Objects.equals(this.played, userItemDataDto.played) &&
        Objects.equals(this.key, userItemDataDto.key) &&
        Objects.equals(this.itemId, userItemDataDto.itemId) &&
        Objects.equals(this.serverId, userItemDataDto.serverId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(rating, playedPercentage, unplayedItemCount, playbackPositionTicks, playCount, isFavorite, lastPlayedDate, played, key, itemId, serverId);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class UserItemDataDto {\n");
    
    sb.append("    rating: ").append(toIndentedString(rating)).append("\n");
    sb.append("    playedPercentage: ").append(toIndentedString(playedPercentage)).append("\n");
    sb.append("    unplayedItemCount: ").append(toIndentedString(unplayedItemCount)).append("\n");
    sb.append("    playbackPositionTicks: ").append(toIndentedString(playbackPositionTicks)).append("\n");
    sb.append("    playCount: ").append(toIndentedString(playCount)).append("\n");
    sb.append("    isFavorite: ").append(toIndentedString(isFavorite)).append("\n");
    sb.append("    lastPlayedDate: ").append(toIndentedString(lastPlayedDate)).append("\n");
    sb.append("    played: ").append(toIndentedString(played)).append("\n");
    sb.append("    key: ").append(toIndentedString(key)).append("\n");
    sb.append("    itemId: ").append(toIndentedString(itemId)).append("\n");
    sb.append("    serverId: ").append(toIndentedString(serverId)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: SampleCode/RestApi/Clients/Net.RestSharp/EmbyClient.Dotnet/Model/UserItemDataDto.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// Class UserItemDataDto  
    /// </summary>
    [DataContract]
        public partial class UserItemDataDto :  IEquatable<UserItemDataDto>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="UserItemDataDto" /> class.
        /// </summary>
        /// <param name="rating">The rating..</param>
        /// <param name="playedPercentage">The played percentage..</param>
        /// <param name="unplayedItemCount">The unplayed item count..</param>
        /// <param name="playbackPositionTicks">The playback position ticks..</param>
        /// <param name="playCount">The play count..</param>
        /// <param name="isFavorite">A value indicating whether this instance is favorite..</param>
        /// <param name="lastPlayedDate">The last played date..</param>
        /// <param name="played">A value indicating whether this &#x60;MediaBrowser.Model.Dto.UserItemDataDto&#x60; is played..</param>
        /// <param name="key">The key..</param>
        /// <param name="itemId">The item identifier..</param>
        /// <param name="serverId">Used only by our Windows app. Not used by Emby Server..</param>
        public UserItemDataDto(double? rating = default(double?), double? playedPercentage = default(double?), int? unplayedItemCount = default(int?), long? playbackPositionTicks = default(long?), int? playCount = default(int?), bool? isFavorite = default(bool?), DateTimeOffset? lastPlayedDate = default(DateTimeOffset?), bool? played = default(bool?), string key = default(string), string itemId = default(string), string serverId = default(string))
        {
            this.Rating = rating;
            this.PlayedPercentage = playedPercentage;
            this.UnplayedItemCount = unplayedItemCount;
            this.PlaybackPositionTicks = playbackPositionTicks;
            this.PlayCount = playCount;
            this.IsFavorite = isFavorite;
            this.LastPlayedDate = lastPlayedDate;
            this.Played = played;
            this.Key = key;
            this.ItemId = itemId;
            this.ServerId = serverId;
        }
        
        /// <summary>
        /// The rating.
        /// </summary>
        /// <value>The rating.</value>
        [DataMember(Name="Rating", EmitDefaultValue=false)]
        public double? Rating { get; set; }

        /// <summary>
        /// The played percentage.
        /// </summary>
        /// <value>The played percentage.</value>
        [DataMember(Name="PlayedPercentage", EmitDefaultValue=false)]
        public double? PlayedPercentage { get; set; }

        /// <summary>
        /// The unplayed item count.
        /// </summary>
        /// <value>The unplayed item count.</value>
        [DataMember(Name="UnplayedItemCount", EmitDefaultValue=false)]
        public int? UnplayedItemCount { get; set; }

        /// <summary>
        /// The playback position ticks.
        /// </summary>
        /// <value>The playback position ticks.</value>
        [DataMember(Name="PlaybackPositionTicks", EmitDefaultValue=false)]
        public long? PlaybackPositionTicks { get; set; }

        /// <summary>
        /// The play count.
        /// </summary>
        /// <value>The play count.</value>
        [DataMember(Name="PlayCount", EmitDefaultValue=false)]
        public int? PlayCount { get; set; }

        /// <summary>
        /// A value indicating whether this instance is favorite.
        /// </summary>
        /// <value>A value indicating whether this instance is favorite.</value>
        [DataMember(Name="IsFavorite", EmitDefaultValue=false)]
        public bool? IsFavorite { get; set; }

        /// <summary>
        /// The last played date.
        /// </summary>
        /// <value>The last played date.</value>
        [DataMember(Name="LastPlayedDate", EmitDefaultValue=false)]
        public DateTimeOffset? LastPlayedDate { get; set; }

        /// <summary>
        /// A value indicating whether this &#x60;MediaBrowser.Model.Dto.UserItemDataDto&#x60; is played.
        /// </summary>
        /// <value>A value indicating whether this &#x60;MediaBrowser.Model.Dto.UserItemDataDto&#x60; is played.</value>
        [DataMember(Name="Played", EmitDefaultValue=false)]
        public bool? Played { get; set; }

        /// <summary>
        /// The key.
        /// </summary>
        /// <value>The key.</value>
        [DataMember(Name="Key", EmitDefaultValue=false)]
        public string Key { get; set; }

        /// <summary>
        /// The item identifier.
        /// </summary>
        /// <value>The item identifier.</value>
        [DataMember(Name="ItemId", EmitDefaultValue=false)]
        public string ItemId { get; set; }

        /// <summary>
        /// Used only by our Windows app. Not used by Emby Server.
        /// </summary>
        /// <value>Used only by our Windows app. Not used by Emby Server.</value>
        [DataMember(Name="ServerId", EmitDefaultValue=false)]
        public string ServerId { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class UserItemDataDto {\n");
            sb.Append("  Rating: ").Append(Rating).Append("\n");
            sb.Append("  PlayedPercentage: ").Append(PlayedPercentage).Append("\n");
            sb.Append("  UnplayedItemCount: ").Append(UnplayedItemCount).Append("\n");
            sb.Append("  PlaybackPositionTicks: ").Append(PlaybackPositionTicks).Append("\n");
            sb.Append("  PlayCount: ").Append(PlayCount).Append("\n");
            sb.Append("  IsFavorite: ").Append(IsFavorite).Append("\n");
            sb.Append("  LastPlayedDate: ").Append(LastPlayedDate).Append("\n");
            sb.Append("  Played: ").Append(Played).Append("\n");
            sb.Append("  Key: ").Append(Key).Append("\n");
            sb.Append("  ItemId: ").Append(ItemId).Append("\n");
            sb.Append("  ServerId: ").Append(ServerId).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as UserItemDataDto);
        }

        /// <summary>
        /// Returns true if UserItemDataDto instances are equal
        /// </summary>
        /// <param name="input">Instance of UserItemDataDto to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(UserItemDataDto input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.Rating == input.Rating ||
                    (this.Rating != null &&
                    this.Rating.Equals(input.Rating))
                ) && 
                (
                    this.PlayedPercentage == input.PlayedPercentage ||
                    (this.PlayedPercentage != null &&
                    this.PlayedPercentage.Equals(input.PlayedPercentage))
                ) && 
                (
                    this.UnplayedItemCount == input.UnplayedItemCount ||
                    (this.UnplayedItemCount != null &&
                    this.UnplayedItemCount.Equals(input.UnplayedItemCount))
                ) && 
                (
                    this.PlaybackPositionTicks == input.PlaybackPositionTicks ||
                    (this.PlaybackPositionTicks != null &&
                    this.PlaybackPositionTicks.Equals(input.PlaybackPositionTicks))
                ) && 
                (
                    this.PlayCount == input.PlayCount ||
                    (this.PlayCount != null &&
                    this.PlayCount.Equals(input.PlayCount))
                ) && 
                (
                    this.IsFavorite == input.IsFavorite ||
                    (this.IsFavorite != null &&
                    this.IsFavorite.Equals(input.IsFavorite))
                ) && 
                (
                    this.LastPlayedDate == input.LastPlayedDate ||
                    (this.LastPlayedDate != null &&
                    this.LastPlayedDate.Equals(input.LastPlayedDate))
                ) && 
                (
                    this.Played == input.Played ||
                    (this.Played != null &&
                    this.Played.Equals(input.Played))
                ) && 
                (
                    this.Key == input.Key ||
                    (this.Key != null &&
                    this.Key.Equals(input.Key))
                ) && 
                (
                    this.ItemId == input.ItemId ||
                    (this.ItemId != null &&
                    this.ItemId.Equals(input.ItemId))
                ) && 
                (
                    this.ServerId == input.ServerId ||
                    (this.ServerId != null &&
                    this.ServerId.Equals(input.ServerId))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.Rating != null)
                    hashCode = hashCode * 59 + this.Rating.GetHashCode();
                if (this.PlayedPercentage != null)
                    hashCode = hashCode * 59 + this.PlayedPercentage.GetHashCode();
                if (this.UnplayedItemCount != null)
                    hashCode = hashCode * 59 + this.UnplayedItemCount.GetHashCode();
                if (this.PlaybackPositionTicks != null)
                    hashCode = hashCode * 59 + this.PlaybackPositionTicks.GetHashCode();
                if (this.PlayCount != null)
                    hashCode = hashCode * 59 + this.PlayCount.GetHashCode();
                if (this.IsFavorite != null)
                    hashCode = hashCode * 59 + this.IsFavorite.GetHashCode();
                if (this.LastPlayedDate != null)
                    hashCode = hashCode * 59 + this.LastPlayedDate.GetHashCode();
                if (this.Played != null)
                    hashCode = hashCode * 59 + this.Played.GetHashCode();
                if (this.Key != null)
                    hashCode = hashCode * 59 + this.Key.GetHashCode();
                if (this.ItemId != null)
                    hashCode = hashCode * 59 + this.ItemId.GetHashCode();
                if (this.ServerId != null)
                    hashCode = hashCode * 59 + this.ServerId.GetHashCode();
                return hashCode;
            }
        }

    }
}
